src/robot/model/testsuite.py

```diff
@@ -40,8 +44,11 @@ class TestSuite(ModelObject):
     __slots__ = ['parent', '_name', 'doc', '_setup', '_teardown',  'rpa',
                  '_my_visitors']
 
-    def __init__(self, name: str = '', doc: str = '', metadata: dict = None,
-                 source: Path = None, rpa: bool = False, parent: 'TestSuite' = None):
+    def __init__(self, name: str = '', doc: str = '',
+                 metadata: Union[Metadata, None] = None,
+                 source: Union[Path, None] = None,
+                 rpa: bool = False,
+                 parent: Union["TestSuite", None] = None):
```

Comment:
- `metadata` can also be `dict` and  `source` can also be `str`.
- Using stringly types like `'Metadata | dict | None'` would probably be better than using `Union[Metadata, dict, None]`.
- Please use single quotes instead of double quotes.
- We need to decide how to handle long signatures like this one. Splitting all parameters to own lines would be a simple solution, but it should be done consistently (i.e. also `name` and `doc` on own lines) and such formatting takes lot of vertical space. I'd try having multiple params per line, but we can make a final decision a bit later. Notice that we use 88 as the maximum line length.